src/path_tracking_pid_nav2_controller.cpp

```diff
@@ -334,6 +338,22 @@ bool PathTrackingPid::computeVelocityCommands(geometry_msgs::msg::TwistStamped&
 {
   RCLCPP_DEBUG(node_->get_logger(), "METHOD CHECK: start of computeVelMethod"); //DEBUG
 
+  if (pid_controller_.getConfig().use_mpc)
+  {
+    predicted_plan_ = pid_controller_.getPredictedPlan();
```

Comment:
does the variables needs to be a global variable? if not make it a local variable defined right above where it is used